src/Compatibility/Microsoft.DotNet.ApiCompat.Task/ValidatePackageTask.cs

```diff
@@ -0,0 +1,106 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.Build.Framework;
+using Microsoft.NET.Build.Tasks;
+using Microsoft.DotNet.ApiCompatibility.Logging;
+
+namespace Microsoft.DotNet.ApiCompat.Task
+{
+    public class ValidatePackageTask : TaskBase
+    {
+        [Required]
+        public string? PackageTargetPath { get; set; }
+
+        [Required]
+        public string? RoslynAssembliesPath { get; set; }
+
+        public string? RuntimeGraph { get; set; }
+
+        public string? NoWarn { get; set; }
+
+        public bool RunApiCompat { get; set; }
+
+        public bool EnableStrictModeForCompatibleTfms { get; set; }
+
+        public bool EnableStrictModeForCompatibleFrameworksInPackage { get; set; }
+
+        public string? BaselinePackageTargetPath { get; set; }
+
+        public bool GenerateCompatibilitySuppressionFile { get; set; }
+
+        public string? CompatibilitySuppressionFilePath { get; set; }
+
+        public ITaskItem[]? PackageAssemblyReferences { get; set; }
+
+        public ITaskItem[]? BaselinePackageAssemblyReferences { get; set; }
+
+        public override bool Execute()
+        {
+            RoslynResolver.Register(RoslynAssembliesPath);
```

Comment:
NIT: this common registration could be abstracted to a new base class if we wanted to.

Reply:
Right. I considered that but I chose this model over a new base class.